src/Adapter/Cart/QueryHandler/GetCartInformationHandler.php

```diff
@@ -120,16 +123,24 @@ public function handle(GetCartInformation $query): CartInformation
 
         $legacySummary = $cart->getSummaryDetails(null, true);
         $addresses = $this->getAddresses($cart);
+        $cartRules = $this->extractCartRulesFromLegacySummary($legacySummary, $currency);
+        $hasFreeShippingCartRule = false;
+        foreach ($legacySummary['discounts'] as $discount) {
+//            if ((bool) $discount[OrderDiscountType::FREE_SHIPPING] && $discount['name'] != self::FREE_SHIPPING_NAME) {
+            if ((bool) $discount[OrderDiscountType::FREE_SHIPPING]) {
```

Comment:
```suggestion
            if ((bool) $discount['free_shipping']) {
```

You shouldn't use `OrderDiscountType::FREE_SHIPPING` because it is related to the cart rule type Although it does match the `free_shipping` property, it's preferable to use the table field hard coded here